Fix the matching rule of instance ProviderID

FengyunPan opened the pull request:
Url.Parse() can't parse ProviderID which contains ':///'.
This PR use regexp to match ProviderID.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
Fix #49769

**Release note**:
```release-note
NONE
```

m1093782566 commented on pkg/cloudprovider/providers/openstack/openstack_test.go:
```diff
@@ -557,17 +557,22 @@ func TestInstanceIDFromProviderID(t *testing.T) {
 	}{
 		{
-			providerID: "openstack://7b9cf879-7146-417c-abfd-cb4272f0c935",
+			providerID: ProviderName + "://" + "/" + "7b9cf879-7146-417c-abfd-cb4272f0c935",
```
I suggest don't remove existing test cases.

FengyunPan replied:
Yes, I agree, the second test case is origin test case:
providerID: ProviderName + "://" + "7b9cf879-7146-417c-abfd-cb4272f0c935",
is the same as "providerID: "openstack://7b9cf879-7146-417c-abfd-cb4272f0c935","

Commit:
Fix the matching rule of instance ProviderID

Url.Parse() can't parse ProviderID which contains ':///'.
This PR use regexp to match ProviderID.

dims commented:
/ok-to-test

dims commented:
@FengyunPan When does this code kick in? who creates the "bad" string, can you figure that out? how did you recreate the problem?

FengyunPan commented:
@dims Please take a look: https://github.com/kubernetes/kubernetes/pull/47593

dims commented:
Thanks for the context @FengyunPan Looks good to me, will wait for CI

FengyunPan commented:
@anguslees @dims  @NickrenREN  - any thoughts on this?

NickrenREN commented:
/approve

FengyunPan commented:
/assign @dims PTAL again.

FengyunPan commented:
/assign @dims
PTAL again.

dims commented:
/approve
/lgtm
